ssl/record/methods/tls_common.c

```diff
@@ -981,6 +982,23 @@ int tls_default_validate_record_header(OSSL_RECORD_LAYER *rl, SSL3_RECORD *rec)
     return 1;
 }
 
+static int tls_do_compress(OSSL_RECORD_LAYER *rl, SSL3_RECORD *wr)
+{
+#ifndef OPENSSL_NO_COMP
+    int i;
+
+    i = COMP_compress_block(rl->compctx, wr->data,
+                            (int)(wr->length + SSL3_RT_MAX_COMPRESSED_OVERHEAD),
+                            wr->input, (int)wr->length);
+    if (i < 0)
+        return 0;
+
+    wr->length = i;
+    wr->input = wr->data;
+#endif
+    return 1;
```

Comment:
Is this right return value? Can this be ever called with OPENSSL_NO_COMP?

Reply:
In practice it shouldn't make a difference since it can't be called in the case of `OPENSSL_NO_COMP`. However if something went wrong and it ended up being called anyway then it would be better if we default to return of 0 - in the same way that we do for `tls_do_uncompress`. I've fixed the code to do it that way.